pisa/core/distribution_maker.py

```diff
@@ -87,17 +99,71 @@ def __init__(self, pipelines, label=None):
             if not isinstance(pipeline, Pipeline):
                 pipeline = Pipeline(pipeline)
             self._pipelines.append(pipeline)
+
+        data_run_livetime = None
+        if set_livetime_from_data:
+            #
+            # Get livetime metadata if defined in any stage in any pipeline
+            #
+            for pipeline_idx, pipeline in enumerate(pipelines):
+                for stage_idx, stage in enumerate(pipeline):
+                    if not (
+                        hasattr(stage, "metadata")
+                        and isinstance(pipeline.metadata, Mapping)
+                        and "livetime" in pipeline.metadata
+                    ):
+                        continue
+
+                    if data_run_livetime is None:
+                        data_run_livetime = stage.metadata["livetime"]
+
+                    if stage.metadata["livetime"] != data_run_livetime:
+                        raise ValueError(
+                            "Pipeline index {}, stage index {} has data"
+                            " livetime = {}, in disagreement with"
+                            " previously-found livetime = {}".format(
+                                pipeline_idx,
+                                stage_idx,
+                                stage.metadata["livetime"],
+                                data_run_livetime,
+                            )
+                        )
+            #
+            # Set param `params.livetime` for any pipelines that have it
+            #
+            if data_run_livetime is not None:
+                data_run_livetime *= ureg.sec
+                for pipeline_idx, pipeline in enumerate(pipelines):
+                    if "livetime" not in pipeline.params.names:
+                        continue
+
+                    pipeline.params.livetime.fixed = True
+                    if pipeline.params.livetime != data_run_livetime:
+                        logging.warning(
+                            "Pipeline index %d has params.livetime = %f, in"
+                            " disagreement with data livetime = %f defined by"
+                            " data. The pipeline's livetime param will be"
+                            " reset to the latter value and set to be fixed"
+                            " (if it is not alredy).",
+                            pipeline_idx,
+                            pipeline.params.livetime,
+                            data_run_livetime,
+                        )
+                        pipeline.params.livetime = data_run_livetime
```

Comment:
Here's an example I just put together, and you can find the example config in the pisa source

```python
from pisa import ureg
from pisa.core.pipeline import Pipeline
from pisa.core.distribution_maker import DistributionMaker

# I'm using the same config twice just for convenience
p0 = Pipeline("~/src/pisa/pisa_examples/resources/settings/pipeline/example.cfg")
p1 = Pipeline("~/src/pisa/pisa_examples/resources/settings/pipeline/example.cfg")

print(p0.params.livetime)

dm = DistributionMaker([p0, p1])

# this does give you all params from all pipelines, but without duplicating duplicated param names
all_params = dm.params
print(all_params.livetime)

# so if you set a param in that returned ParamSet, it doesn't set _all_ params in _all_ contained pipelines.
# This is probably bug since a user might expect for that to be the case:
all_params.livetime = 3 * ureg.common_year

# thus this fails (as the following retrieves the params afresh from each pipeline
# and checks if the params of same name have same value in each pipeline)
print(dm.params.livetime)

# instead, one still has to set the param on each pipeline individually
for pipeline in dm:
    if "livetime" in pipeline.params.names:
        pipeline.params.livetime = 3 * ureg.common_year
        pipeline.params.livetime.is_fixed = True

print(dm.params.livetime)
````
You'll note that I probably have another bug, my PR's code has `fixed = True` as the attr being set, but it should be `is_fixed = True`